File: php/src/Advertisement/UI/Http/RenewAdvertisementController.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisement\UI\Http;

use Demo\App\Advertisement\Application\Command\RenewAdvertisement\RenewAdvertisementCommand;
use Demo\App\Advertisement\Application\Command\RenewAdvertisement\RenewAdvertisementUseCase;
use Demo\App\Framework\FrameworkRequest;
use Demo\App\Framework\FrameworkResponse;

final class RenewAdvertisementController
{
    public function __construct(private RenewAdvertisementUseCase $useCase)
    {
    }

    public function request(FrameworkRequest $request): FrameworkResponse
    {
        $command = new RenewAdvertisementCommand(
            $request->getIdPath(),
            ($request->content())['password'],
        );

        $this->useCase->execute($command);

        return new FrameworkResponse(
            FrameworkResponse::STATUS_OK,
            []
        );
    }
}

File: kotlin/src/main/kotlin/advertisement/ui/http/RenewAdvertisementController.kt
package advertisement.ui.http

import advertisement.application.renewAdvertisement.RenewAdvertisementCommand
import advertisement.application.renewAdvertisement.RenewAdvertisementUseCase
import framework.FrameworkRequest
import framework.FrameworkResponse

class RenewAdvertisementController(private val useCase: RenewAdvertisementUseCase) {

    fun execute(request: FrameworkRequest): FrameworkResponse {
        useCase.execute(
            RenewAdvertisementCommand(
                request.getIdPath(),
                request.content["password"]!!,
            )
        )

        return FrameworkResponse(
            FrameworkResponse.STATUS_OK,
            mapOf(),
        )
    }
}